src/components/viewSeminars/index.vue

```diff
@@ -471,34 +463,17 @@ export default {
     }
   },
   methods: {
-    // onManualFilterChange() {
-    //   Object.keys(this.SUGGESTED_SEARCH_STATE).forEach(
-    //     key => (this.SUGGESTED_SEARCH_STATE[key] = FALSE)
-    //   );
-    // },;
-    handleCheckOpenOnly(val) {
-      console.log(val);
-      console.log(this.filters.openOnly);
-    },
-    checkedLecturesOnly(data) {
-      console.log(data.target.checked);
-    },
     onSuggestedSearchSelectToggle(data) {
       // Update which button is selected.
       const new_button_state = {};
-      Object.keys(this.SUGGESTED_SEARCH_STATE).forEach(key => {
-        new_button_state[key] = false;
+      _.forOwn(this.SUGGESTED_SEARCH_STATE, (val, key) => {
+        if (key === data.id) {
+          new_button_state[key] = data.value;
+        } else {
+          new_button_state[key] = false;
+        }
       });
-      if (data.value === true) {
-        new_button_state[data.id] = true;
-      }
       this.SUGGESTED_SEARCH_STATE = new_button_state;
-      // Update filter values
-      const new_filters = { ...DEFAULT_FILTERS };
-      if (data.value) {
-        Object.assign(new_filters, SUGGESTED_SEARCH_FILTERS[data.id]);
-      }
-      this.filters = new_filters;
```

Comment:
I move all effects of `this.SUGGESTED_SEARCH_STATE` on `this.filters` from here to `watch`, so that the changes caused by `watch.filters()` on `this.SUGGESTED_SEARCH_STATE` will be processed in the same place.